src/transaction_builder.js

```diff
@@ -422,16 +422,13 @@ function buildInput (input, allowIncomplete) {
     witness = buildStack(bscript.types.P2PKH, input.signatures, input.pubKeys, allowIncomplete)
   } else if (scriptType === bscript.types.P2WSH) {
     // We can remove this check later
-    if (SIGNABLE.indexOf(input.witnessScriptType) !== -1) {
+    if (SIGNABLE.indexOf(input.witnessScriptType) === -1 && !allowIncomplete) {
+      throw new Error('Impossible to sign this type')
+    } else if (SIGNABLE.indexOf(input.witnessScriptType) !== -1) {
       witness = buildStack(input.witnessScriptType, input.signatures, input.pubKeys, allowIncomplete)
       witness.push(input.witnessScript)
-    } else {
-      // We can remove this error later when we have a guarantee prepareInput
-      // rejects unsignble scripts - it MUST be signable at this point.
-      throw new Error()
+      scriptType = input.witnessScriptType
```

Comment:
ignore this comment, the `else if` check above assures `input.witnessScriptType` will be defined